bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusAccountHandler.java

```diff
@@ -120,46 +121,47 @@ public void dispose() {
         if (refreshTokenFuture != null) {
             refreshTokenFuture.cancel(true);
         }
+        super.dispose();
     }
 
-    @Override
-    public void initialize() {
+    private boolean login() {
         GroheOndusAccountConfiguration config = getConfigAs(GroheOndusAccountConfiguration.class);
-
         if (this.accountServlet == null) {
             this.accountServlet = new AccountServlet(httpService, this.getThing().getUID().getId(), this);
         }
 
         if ((config.username == null || config.password == null) && !this.hasRefreshToken()) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
                     "Need username/password or refreshToken");
-            return;
+            return false;
         }
 
-        updateStatus(ThingStatus.UNKNOWN);
         try {
             if (storage.containsKey(STORAGE_KEY_REFRESH_TOKEN)) {
                 ondusService = OndusService.login(storage.get(STORAGE_KEY_REFRESH_TOKEN));
-                scheduleTokenRefresh();
             } else {
                 // TODO: That's probably really inefficient, internally the loginWebform method acquires a refresh
                 // token, maybe there should be a way to obtain this token here, somehow.
                 ondusService = OndusService.loginWebform(config.username, config.password);
             }
+            scheduleTokenRefresh();
             updateStatus(ThingStatus.ONLINE);
-
-            scheduler.submit(() -> getThing().getThings().forEach(thing -> {
-                GroheOndusBaseHandler thingHandler = (GroheOndusBaseHandler) thing.getHandler();
-                if (thingHandler != null) {
-                    thingHandler.updateChannels();
-                }
-            }));
+            return true;
         } catch (LoginException e) {
             logger.debug("Grohe api login failed", e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Login failed");
```

Comment:
Consider moving text to src/main/resources/OH-INF/i18n/groheondus.properties to allow for localization.